21/day01/src/main.rs

```diff
@@ -0,0 +1,41 @@
+fn main() {
+    let i = include_str!("input.txt");
+    let mut s = i.split('\n');
+    let mut x = s.next().unwrap();
+    let mut y = s.next().unwrap();
+    let mut c = 0;
+    loop {
+        if x.parse::<i32>().unwrap() < y.parse::<i32>().unwrap() {
+            c += 1;
+        }
+        x = y;
+        let n = s.next();
+        match n {
+            None => break,
+            Some(some_n) => y = some_n,
+        }
+    }
+    println!("{}", c);
+    s = i.split('\n');
+    x = s.next().unwrap();
+    y = s.next().unwrap();
+    let mut z = s.next().unwrap();
+    let mut zz = s.next().unwrap();
```

Comment:
create a separate function for your parser.

Reply:
Well done. But unfortunately not what I meant. Now include in this function all operations which are needed for the input information well prepared. Also rename your function in a meaningful way.

Reply:
I'm sorry, I don't understand this.